File: backend/src/main/java/org/schoolmela/quiz/model/QuizResult.java
package org.schoolmela.quiz.model;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "quiz_results")
public class QuizResult {
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Quiz getQuiz() {
        return quiz;
    }

    public void setQuiz(Quiz quiz) {
        this.quiz = quiz;
    }

    public int getScore() {
        return score;
    }

    public void setScore(int score) {
        this.score = score;
    }

    public int getTotalQuestions() {
        return totalQuestions;
    }

    public void setTotalQuestions(int totalQuestions) {
        this.totalQuestions = totalQuestions;
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne
    @JoinColumn(name = "quiz_id", nullable = false)
    private Quiz quiz;

    @Column(nullable = false)
    private int score;

    @Column(nullable = false)
    private int totalQuestions;
}

File: flutter/lib/models/quiz_result.dart
import 'quiz.dart';

class QuizResult {
  final int id;
  final Quiz quiz;
  final int score;
  final int totalQuestions;

  QuizResult({
    required this.id,
    required this.quiz,
    required this.score,
    required this.totalQuestions,
  });

  factory QuizResult.fromJson(Map<String, dynamic> json) {
    return QuizResult(
      id: json['id'],
      quiz: Quiz.fromJson(json['quiz']),
      score: json['score'],
      totalQuestions: json['totalQuestions'],
    );
  }
}